### DISCUSSION TITLE
@computed() and Dirve.getUrl()

### DISCUSSION THREAD

**Discussion by guoyunhe** [adonisjs/core#4048] [Help] [answered]
I have an Image model which I store storage key as `path` attribute. I also want to get `url` computed attribute. However, `Drive.getUrl()` is an async function, and cannot be used in a getter:

```ts
export default class Image extends Model {
  /** File storage path, images/<md5> */
  @column()
  public path: string;

  /** Image public url */
  @computed()
  public get url() {
    return Drive.getUrl(this.path)
  }
}
```

How could I do it?

**Comment by steveyout** [comment, marked as answer]
use ```@afterFetch()``` or ```afterFind``` hooks
